tools/probes/probes_main.c

```diff
@@ -0,0 +1,96 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2018 Intel Corporation. All rights reserved.
+//
+// Author: Adrian Bonislawski <adrian.bonislawski@intel.com>
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <stdint.h>
+
+#include "wave.h"
+
+#define APP_NAME "sof-probes"
+
+#define DEBUGFS_DIR "/sys/kernel/debug/sof"
+#define WAVE_DIR "/"
+
+static const char * const debugfs[] = {
+	"probe_points", "probe_points_disconnect", "probe_injection_dma",
+};
+
+/* this will be included from sof probe files */
+#define SYNC_WORD 0xBABEBEBA
+
+struct probe_data_packet {
+	uint32_t sync_word;
+	uint32_t buffer_id;
+	uint32_t format;
+	uint32_t timestamp_low;
+	uint32_t timestamp_high;
+	uint32_t data_size_bytes;
+	uint32_t data[];
+};
+
+/* --- */
+
+enum connection_purpose {
+	EXTRACT = 0,
+	INJECT,
+	INJECT_REEXTRACT
+};
+
+static void usage(void)
+{
+	fprintf(stdout, "Usage %s <option(s)> <buffer_id/file>\n\n", APP_NAME);
+	fprintf(stdout, "%s:\t -e buffer_id\tConnect extraction probe\n", APP_NAME);
+	fprintf(stdout, "%s:\t -i buffer_id\tConnect injection probe\n", APP_NAME);
+	fprintf(stdout, "%s:\t -d buffer_id\tDisconnect probe from buffer\n\n", APP_NAME);
+	fprintf(stdout, "%s:\t -p file\tParse extracted file\n\n", APP_NAME);
+	fprintf(stdout, "%s:\t -h \t\tHelp, usage info\n", APP_NAME);
+	exit(0);
+}
+
+void connect_probe(uint32_t buffer_id, enum connection_purpose purpose)
```

Comment:
I would also expect these to pass some private data structure with context.

Reply:
These connect and disconnect functions are very small and dont need any structures. I will pass some structures to main parse function when some config options will be added later